### PR TITLE
Make configFetcher fallback work for gitea

### LINKED ISSUE #133: Use .woodpecker.yml as default pipeline config file
As the current .drone.yml files are not supported by woodpecker, it seems to useful to switch to  supporting .woodpecker.yml as a default configuration source

### PR DESCRIPTION
**Pull request by 6543**
#299 who closed #133 did not take into account, that that gitea (and eventually) other forges do return 200 and empty string if file was not found - this make configFetcher more resilient

### CODE REVIEW

File: server/shared/configFetcher.go [hunk 1]
```diff
@@ -54,25 +55,29 @@ func (cf *configFetcher) Fetch() (files []*remote.FileMeta, err error) {
...
+				if err == nil && len(file) != 0 {
 					return []*remote.FileMeta{{
 						Name: ".drone.yml",
 						Data: file,
 					}}, nil
 				}
+
+				if err == nil && len(files) == 0 {
+					return nil, fmt.Errorf("ConfigFetcher: Fallback did not found config")
+				}
```
Comment (anbraten) [inline server/shared/configFetcher.go:80]:
```suggestion
			  return nil, fmt.Errorf("ConfigFetcher: Fallback did not found config")
```

Shouldn't this be enough?

Comment (6543) [reply server/shared/configFetcher.go:80]: well then we would drop the err if there is one

Comment (6543) [reply server/shared/configFetcher.go:80]: if we simplify we otherwise would have to drop retry func

Comment (anbraten) [reply server/shared/configFetcher.go:80]: > if we simplify we otherwise would have to drop retry func

Upps, we currently don't use the retry anyways as of `return nil, err` in line 83.

Adding some separate tests for empty directories and empty file contents would be nice.

File: server/shared/configFetcher_test.go [hunk 2]
```diff
@@ -192,22 +247,11 @@ func TestFetch(t *testing.T) {
...
-				var receivedFileNames []string
-				for _, file := range files {
-					receivedFileNames = append(receivedFileNames, file.Name)
-				}
-				t.Fatal("expected some other pipeline files", tt.expectedFileNames, receivedFileNames)
+			matchingFiles := make([]string, len(files))
+			for i := range files {
+				matchingFiles[i] = files[i].Name
 			}
+			assert.ElementsMatch(t, tt.expectedFileNames, matchingFiles, "expected some other pipeline files")
```
Comment (anbraten) [inline server/shared/configFetcher_test.go:254]: nice improvement 😉

File: server/shared/configFetcher.go [hunk 3]
```diff
@@ -54,25 +55,30 @@ func (cf *configFetcher) Fetch() (files []*remote.FileMeta, err error) {
...
 			}
 
 			return nil, err
```
Comment (anbraten) [inline server/shared/configFetcher.go:84]: I think this line is stopping the retry loop. Should we just remove the retry loop?

Comment (6543) [reply server/shared/configFetcher.go:84]: can we leave it for now?

I like to add logging (more precice one!) - if we do have that we can drop it

Comment (anbraten) [reply server/shared/configFetcher.go:84]: We can just leave it as it is now and add a TODO, saying the retry loop is inactive and could maybe be deleted in future.

Comment (6543) [inline server/shared/configFetcher.go:84]:
```suggestion
			// TODO: retry loop is inactive and could maybe be fixed/deleted
			return nil, err
```

**Review by anbraten — APPROVED** [review]
